pkg/machinery/config/merge/merge.go

```diff
@@ -102,11 +106,32 @@ func merge(vl, vr reflect.Value, replace bool) error {
 
 		for _, k := range vr.MapKeys() {
 			if vl.MapIndex(k) != zeroValue {
-				v := reflect.New(tl.Elem()).Elem()
-				v.Set(vl.MapIndex(k))
+				valueType := tl.Elem()
+
+				var v, rightV reflect.Value
+
+				if valueType.Kind() == reflect.Interface { // special case for map[string]interface{}
+					// here we have to instantiate the actual type behind in the `interface{}`, otherwise it's not settable
+					valueType = vl.MapIndex(k).Elem().Type()
+
+					if valueType != vl.MapIndex(k).Elem().Type() {
```

Comment:
Isn't valueType set from vl.MapIndex(k).Elem().Type()?

Reply:
oops, it should have been `vr` there, nice catch